src/background/services/openPayments.ts

```diff
@@ -904,16 +919,51 @@ export class OpenPaymentsService {
     );
   }
 
+  async manualReconnectWallet() {
+    try {
+      await this.rotateToken();
+    } catch (error) {
+      if (isInvalidClientError(error)) {
+        const msg = this.t('connectWallet_error_invalidClient');
+        throw new Error(msg, { cause: error });
+      }
+      throw error;
+    }
+    await this.storage.setState({ key_revoked: false });
+  }
+
   async reconnectWallet() {
     try {
+      const { walletAddress } = await this.storage.get(['walletAddress']);
+      if (
+        !isWalletAddress(walletAddress) ||
+        !KeyAutoAddService.supports(walletAddress)
+      ) {
+        // this.updateConnectStateError(error);
+        throw new ErrorWithKey('connectWalletKeyService_error_notImplemented');
```

Comment:
IMPORTANT: The reason we have this in `connectWallet` like that (i.e. check if `autoKeyAddConsent`) can be found in https://github.com/interledger/web-monetization-extension/pull/637#discussion_r1787599150. Likely worth following same approach here.